src/back/wgsl/writer.rs

```diff
@@ -491,11 +480,17 @@ impl<W: Write> Writer<W> {
                         if multi { "multisampled_" } else { "" },
                         String::from(""),
                     ),
-                    Ic::Storage(storage_format) => (
-                        "storage_",
-                        "",
-                        format!("<{}>", storage_format_str(storage_format)),
-                    ),
+                    Ic::Storage { format, access } => {
+                        if access.contains(crate::StorageAccess::STORE) {
+                            (
+                                "storage_",
+                                "",
+                                format!("<{},write>", storage_format_str(format)),
```

Comment:
read_write is missing

Reply:
ATM the [spec](https://gpuweb.github.io/gpuweb/wgsl/#texture-storage) says:

"access must be write, if specified."

The grammar suggests otherwise:

```
texture_sampler_types
  : sampler_type
  | depth_texture_type
  | sampled_texture_type LESS_THAN type_decl GREATER_THAN
  | multisampled_texture_type LESS_THAN type_decl GREATER_THAN
  | storage_texture_type LESS_THAN texel_format COMMA access_mode GREATER_THAN
  ```

where access_mode is:

```
access_mode
  : READ
  | WRITE
  | READ_WRITE
```

Are you suggesting Naga disregards that statement and just implements the grammar as-is?

Reply:
No, I think we should follow up the spec.